src/payment/CardDetails.jsx

```diff
@@ -41,13 +46,32 @@ export class CardDetailsComponent extends React.Component {
     return items;
   }
 
-  updateCardType = (event, newValue) => {
+  handleCardNumberChange = (event, newValue) => {
     let cardIcon = null;
+    let cardId = '';
     const { card } = CardValidator.number(newValue);
     if (card) {
-      cardIcon = SUPPORTED_CARD_ICONS[card.type];
+      const cardInfo = SUPPORTED_CARDS[card.type];
+      cardId = cardInfo.id;
+      cardIcon = cardInfo.icon;
     }
-    this.setState({ cardIcon });
+    this.setState({ cardId, cardIcon, cardNumber: newValue });
+  };
+
+  handleSecurityCodeChange = (event, newValue) => {
+    this.setState({ securityCode: newValue });
+  };
+
+  updateCardExpiryMonth = (event, newValue) => {
+    const cardExpiryParts = this.state.cardExpiryDate.split('-');
+    cardExpiryParts[0] = newValue.padStart(2, '0');
```

Comment:
It looks like `padStart` isn't supported in IE.  It might be worth double-checking that Babel is handling it properly... easy way would be to just search for `padStart` in the built app.js artifact and make sure it's not there. ;)